commands/box/contract/root.go

```diff
@@ -1240,10 +1192,77 @@ func calcGasLimit(bal, amount uint64) uint64 {
 	return limit
 }
 
-func isHexData(str string) bool {
-	_, err := hex.DecodeString(str)
-	if err != nil {
+func isHexFormat(str string) bool {
+	if len(str) == 0 {
+		return false
+	}
+	if _, err := hex.DecodeString(str); err != nil {
 		return false
 	}
 	return true
 }
+
+func parseContractData(bytecode string, data string) (string, error) {
+	//parase data
+	if _, err := os.Stat(data); err == nil {
+		bytes, err := ioutil.ReadFile(data)
+		if err != nil {
+			fmt.Println("read contract file error:")
+			return "read contract file error", err
+		}
```

Comment:
if isHexFormat(byte code) {
  // it is probably a contract bin
  
} else {
  // maybe it is a contract source file, try to compile it to bin with abi file
}